ca.go

```diff
@@ -18,19 +20,29 @@ type CA struct {
 	// may be set to something like `nil`.
 	Rand io.Reader
 
-	// Wrapped `crypto.Signer` to preform OpenSSH CA operations.
-	Signer ssh.Signer
+	// Function to choose which signer to use based on request information
+	ChooseSigner func(events.APIGatewayProxyRequest) (crypto.Signer, error)
 }
 
 // Sign an SSH Certificate template (with `Key` set), and return the
 // certificate.
-func (s CA) Sign(template ssh.Certificate) (*ssh.Certificate, error) {
+func (s CA) Sign(template ssh.Certificate, event events.APIGatewayProxyRequest) (*ssh.Certificate, error) {
```

Comment:
I think it'd make more sense for this to take a dedicated struct, instead of the whole event. That way the API surface is narrower and less prone to breaks.

Reply:
Oh, oops! I forgot to address this, let me work on an update.